src/Evaluator.hs

```diff
@@ -23,6 +80,14 @@ import Syntax.SrcLoc
 import Type
 import Value
 
+-- -----------------------------------------------------------------------------
+-- Interpreter Data Types
+
+
+data EvalResult
+  = ExampleResult SrcLoc VEnv Core (Scheme Ty) Value
+  | IOResult (IO ())
+
 -- TODO: more precise representation
 type Type = Text
```

Comment:
Text is actually just fine since the type checker has (hopefully) already caught all the type errors, so it doesn't make sense to spend extra effort on type errors which slip through to evaluation-time.